### DISCUSSION TITLE
A quick (maybe naive) question about feature importance.

### DISCUSSION THREAD

**Discussion by jorgpg5** [timeseriesAI/tsai#783] [Q&A] [answered]
Hello, 

Thanks for sharing this awesome work! I have a question regarding how to "read" the results from the feature importance (by permutation) for a classification task. According to the [documentation](https://timeseriesai.github.io/tsai/analysis.html):

"_**The permutation feature or step importance is defined as the decrease in a model score when a single feature or step value is randomly shuffled. So if you using accuracy (higher is better), the most important features or steps will be those with a lower value on the chart (as randomly shuffling them reduces performance).**_"

I'm using accuracy as the metric. Thereby, if the values of the features are smaller on the graph, these features are more important for the model. For instance, var_4, var_6, var_7, var_23, var28, and var35 might be very relevant to the model due to the small values. Is that right? However, what happens when we have negative/red values? I have attached a figure of where this happens. Thank you so much for your help and awesome work! 

![feat_importance_example](https://github.com/timeseriesAI/tsai/assets/70129680/a4827357-ee0a-4931-9912-32db24dd344a)

**Comment by vrodriguezf** [comment, marked as answer]
I might be wrong, but the way I see it is that large green bars represent important features (because removing them caused a large accuracy decrease) and red bars represent misleading features (removing them made the accuracy be higher)

**Reply by jorgpg5** [reply]
Thanks for your reply, Victor! I'll wait for a couple more answers before marking your reply as the answer (if that's okay). Thank you!

**Comment by oguiza** [comment]
Hi @jorgpg5, Victor’s answer is correct. Features in red or close to 0 are good candidates to be removed. Yo can drop then and retrain the model again.

**Reply by jorgpg5** [reply]
Awesome! Thank you so much for confirming this, Ignacio. Again, thank you for the great work!
